app/Http/Controllers/Admin/ColumnMonsterCrudController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+
+/**
+ * Class ColumnMonsterCrudController.
+ *
+ * @property-read \Backpack\CRUD\app\Library\CrudPanel\CrudPanel $crud
+ */
+class ColumnMonsterCrudController extends MonsterCrudController
+{
+    /**
+     * Configure the CrudPanel object. Apply settings to all operations.
+     *
+     * @return void
+     */
+    public function setup()
+    {
+        CRUD::setModel(\App\Models\Monster::class);
+        CRUD::setRoute(config('backpack.base.route_prefix').'/column-monster');
+        CRUD::setEntityNameStrings('column monster', 'column monsters');
+        $this->crud->set('show.setFromDb', false);
+    }
+
+    /**
+     * Define what happens when the List operation is loaded.
+     *
+     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
+     *
+     * @return void
+     */
+    public function setupListOperation()
+    {
+        $this->crud->disableResponsiveTable();
+
+        $this->crud->addColumns(static::getFieldsArrayForSimpleTab());
+        $this->crud->addColumns($this::getFieldsArrayForTimeAndSpaceTab());
+        $this->crud->addColumns(static::getFieldsArrayForSelectsTab());
+        $this->crud->addColumns(static::getFieldsArrayForRelationshipsTab());
+        $this->crud->addColumns(static::getFieldsArrayForUploadsTab());
+        $this->crud->addColumns(static::getFieldsArrayForWysiwygEditorsTab());
+        $this->crud->addColumns(static::getFieldsArrayForMiscellaneousTab());
+
+        /**
+         * Columns can be defined using the fluent syntax or array syntax:
+         * - CRUD::column('price')->type('number');
+         * - CRUD::addColumn(['name' => 'price', 'type' => 'number']);.
+         */
+    }
+
+    public static function getFieldsArrayForTimeAndSpaceTab()
```

Comment:
Hmm... why do we still have this function - `getFieldsArrayForTimeAndSpaceTab()`, @munjaldevelopment ? Any reason we're not calling the one in the original MonsterCrudController instead?

Reply:
[](https://github.com/Laravel-Backpack/demo/pull/430#discussion_r990627367)

We are having this error if we call default function as  ['start_date', 'end_date'] name call in array (in date_range). So, I called this function and call both separately.

Reply:
Hmmm what error is that? That error highlights that the `date_range` column syntax is not the same as the `date_range` field syntax. It's exactly why we're doing this, why we're turning fields into columns 😀

Reply:
I have commented "date_range" column definition from columnMonsterCrud & removed "date_range" column blade too.

Reply:
How about the rest of them? 😀 Nope, this entire method has to go away, @munjaldevelopment . 

If there's one column that doesn't work (`date_range`) what we can do is store the content of the ORIGINAL method in a variable, then process that variable to remove that one field that doesn't work. So something like:
 
```php
$spaceTimeFields = static::getFieldsArrayForTimeAndSpaceTab();
// TODO: process $spaceTimeFields to remove any columns you want removed, 
// and say WHY they were removed

        $this->crud->addColumns(static::getFieldsArrayForSimpleTab());
        $this->crud->addColumns($spacetimeFields);
        $this->crud->addColumns(static::getFieldsArrayForSelectsTab());
        $this->crud->addColumns(static::getFieldsArrayForRelationshipsTab());
        $this->crud->addColumns(static::getFieldsArrayForUploadsTab());
        $this->crud->addColumns(static::getFieldsArrayForWysiwygEditorsTab());
        $this->crud->addColumns(static::getFieldsArrayForMiscellaneousTab());
```

Reply:
Removed "date_range" key as suggested